From 0797320a7a8d6a8bd899a4149322486db7f5baa2 Mon Sep 17 00:00:00 2001
From: Koen Kooi <koen@dominion.thruhere.net>
Date: Thu, 22 Sep 2011 11:27:13 +0200
Subject: [PATCH 1/3] analyze: always draw 1s marker for scale

In situations like this:

root@omap4430-panda:~# systemd-analyze
Startup finished in 1499ms (kernel) + 916ms (userspace) = 2416ms

The svg plot will only have the 0s marker and no subsequent markers for scale. This patch forces the 1s marker to always be drawn.

Signed-off-by: Koen Kooi <koen@dominion.thruhere.net>
---
 src/systemd-analyze |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/systemd-analyze b/src/systemd-analyze
index ae7dcfb..649d0e1 100755
--- a/src/systemd-analyze
+++ b/src/systemd-analyze
@@ -147,7 +147,7 @@ elif sys.argv[1] == 'plot':
         context.set_line_width(1)
         context.set_source_rgb(0.7, 0.7, 0.7)
 
-        for x in range(0, (finish_time - start_time)/10000, 100):
+        for x in range(0, max((finish_time - start_time)/10000,110), 100):
                 context.move_to(x, 0)
                 context.line_to(x, height-border*2)
 
@@ -163,7 +163,7 @@ elif sys.argv[1] == 'plot':
         banner = "Running on %s (%s %s) %s" % (os.uname()[1], os.uname()[2], os.uname()[3], os.uname()[4])
         draw_text(context, 0, -15, banner, hcenter = 0, vcenter = 1)
 
-        for x in range(0, (finish_time - start_time)/10000, 100):
+        for x in range(0, max((finish_time - start_time)/10000,110), 100):
                 draw_text(context, x, -5, "%lus" % (x/100), vcenter = 0, hcenter = 0)
 
         y = 0
-- 
1.6.6.1

